frontend/src/components/Toast/Toast.tsx

```diff
@@ -0,0 +1,84 @@
+import {
+  Alert,
+  AlertDescription,
+  AlertIcon,
+  AlertProps,
+  AlertTitle,
+  Box,
+  CloseButton,
+  useMediaQuery,
+  UseToastOptions,
+} from '@chakra-ui/react'
+
+type ToastStatus = 'danger' | 'success' | 'warning'
+
+export interface ToastProps
+  extends Omit<
+    UseToastOptions,
+    'duration' | 'position' | 'render' | 'status' | 'variant'
+  > {
+  /**
+   * The status of the Toast.
+   */
+  status: ToastStatus
+  // RenderProps that chakra passes to all custom components that uses the render function
+  onClose: () => void
+}
+
+const getChakraAlertStatus = (status: ToastStatus): AlertProps['status'] =>
+  status === 'danger' ? 'error' : status
+
+export const Toast = ({
+  status,
+  title,
+  id,
+  description,
+  isClosable,
+  onClose,
+  onCloseComplete,
+}: ToastProps): JSX.Element => {
+  // Dynamically determine the size of the modal based on screen breakpoints
+  const [isDesktop] = useMediaQuery('(min-width: 1080px)')
+  const width = isDesktop ? '680px' : 'auto'
+  const horizontalMargins = isDesktop ? 'inherit' : 2
+
+  return (
+    <Alert
+      status={getChakraAlertStatus(status)}
+      variant="toast"
+      colorScheme={status}
+      padding={4}
+      id={String(id)}
+      // Padding right is 4 rem (normal padding) + width of the button.
+      // This is to prevent the button overlapping the text on resize.
+      pr={10}
```

Comment:
As discussed (probably after you made this PR), all these styling props should go into the theme config instead of being written here

Reply:
i agree and i've made the changes; however, something that i'm not too sure of is if we could shift the `Alert` styling and theming directly into the theme file as well, so that it's not present on the component and exists solely within the theme.

Reply:
Yes, we should. Then we can just call `useStyleConfig`

Reply:
donezo, explanation here for future readers so knowledge is shared: we shouldn't couple css classes together because actually css is _flat_. just because the components have similar naming does not mean that their styles should be tied together. 

rather, parent level components should only handle formatting at the top level and be agnostic of children specific styling.